internal/cmd/docs.go

```diff
@@ -0,0 +1,44 @@
+package cmd
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/cobra/doc"
+	"github.com/spf13/viper"
+)
+
+var (
+	flagDocsMarkdownOutputDir string
+)
+
+// docsMarkdownCmd represents the markdown command
+var docsMarkdownCmd = &cobra.Command{
+	Use:   "markdown",
+	Short: "Generates markdown documentation for havener.",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		var err error
+
+		flagDocsMarkdownOutputDir = viper.GetString("output-dir")
+
+		if flagDocsMarkdownOutputDir, err = absolutePath(flagDocsMarkdownOutputDir); err != nil {
+			return err
+		}
+		return doc.GenMarkdownTree(rootCmd, flagDocsMarkdownOutputDir)
+	},
+}
+
+func init() {
+	rootCmd.AddCommand(docsMarkdownCmd)
```

Comment:
Is there a way we can have this as a separate program next to `github.com/homeport/havener/cmd/havener` so that the markdown generation is not part of `havener`?

Reply:
we can have a docs cmd, and under that the markdown. Is that something better?

Reply:
No, I was more thinking that it would be another standalone command next to `havener`, but one we do not build. As part of a pre-commit hook or as a make target, we would call `go run cmd/mkdown/main.go` and it would replace the current documentation.

Reply:
I see, interesting approach. I will work on it, thanks.